### DISCUSSION TITLE
define location for download

### DISCUSSION THREAD

**Discussion by nauni77** [pyenv/pyenv#3082] [Ideas]
Hello, 

I think about using `pyenv` in our build pipelines. But it's no good idea always downloading the `tar.gz`-file from the internet. If I install a version of python it shows:
```
Downloading Python-3.12.7.tar.xz...
-> https://www.python.org/ftp/python/3.12.7/Python-3.12.7.tar.xz
Installing Python-3.12.7...
```
Is there a possibility to use a different server for download? Something like `https://www.mycomapany.com/python-org/` instead of `https://www.python.org/ftp/python/`. It would be perfect if the own mirror is tried and use the default if it was not found.

Best regards, Oli

**Comment by native-api** [comment]
First, `$PYENV_ROOT/cache`, if it exists, holds downloaded packages. See https://github.com/pyenv/pyenv/blob/master/plugins/python-build/README.md#package-download-cache

Then you indeed can set up a mirror, see https://github.com/pyenv/pyenv/blob/master/plugins/python-build/README.md#package-download-mirrors
